18-websockets/classworks/Alexey_Goremykin/src/pages/Chat/Chat.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import './Chat.css';
+import Message from './components/Message';
+import io from 'socket.io-client';
+const socket = io('http://localhost:3001/', {transports: ['websocket']});
+
+socket.on('connection', () => {
+  console.log(socket.connected);
+});
+
+function Chat() {
+
+  const [message, setMessage] = useState('');
+  const [userName, setUserName] = useState('');
+  const [messages, setMessages] = useState([]);
+  const [typingMessage, setTypingMessage] = useState('')
+
+  const sendMessage = () => {
+    socket.emit('send-message', {userName, message});
+    socket.on('new-message', ([...newMessages]) => {
+      setMessages(newMessages);
+    })
+  }
+
+  const messageValue = (e) => {
+    setMessage(e.target.value)
+    socket.emit('typing-message', userName);
```

Comment:
if the user is a super typer man, you will get a huge number of requests from him. A good practice is to use the debounce method for calling function no more than one time per 2-4 secs.
Unfortunately, javascript doesn't have this method. You can write it by yourself or use some library.